ddtrace/internal/_encoding.pyx

```diff
@@ -166,22 +166,26 @@ cdef class StringTable(object):
     cdef insert(self, object string):
         pass
 
-    cdef stdint.uint32_t _index(self, object string):
+    cdef stdint.uint32_t _index(self, object string) except? -1:
         cdef stdint.uint32_t _id
+        cdef int ret
 
         if string is None:
             return 0
 
-        if PyDict_Contains(self._table, string):
+        ret = PyDict_Contains(self._table, string)
+        if ret == -1: return ret
+        if ret:
             return PyLong_AsLong(<object>PyDict_GetItem(self._table, string))
 
         _id = self._next_id
-        PyDict_SetItem(self._table, string, PyLong_FromLong(_id))
+        ret = PyDict_SetItem(self._table, string, PyLong_FromLong(_id))
+        if ret != 0: return ret
```

Comment:
```suggestion
        if ret == -1: return ret
```

(although seems like != 0 is fine too https://docs.python.org/3/c-api/dict.html#c.PyDict_SetItem)

but we already do `ret == -1` above so worth being consistent imo